src/logger.ts

```diff
@@ -41,23 +41,29 @@ export const debug_info = {
 export async function saveInterfaceStat(): Promise<void> {
   console.log(apiPerfLogData)
   try {
-    // eslint-disable-next-line prefer-const
-    let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[0]
-    // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-    let placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-    let sql = 'INSERT INTO interface_stats VALUES (' + placeholders + ')'
-    for (let i = 1; i < apiPerfLogData.length; i++) {
-      // eslint-disable-next-line prefer-const
-      let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[i] // eslint-disable-line security/detect-object-injection
-
-      // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-      placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-      sql = sql + `, (${placeholders})`
-    }
+    const stmt = db.prepare(`
+      INSERT INTO interface_stats 
+      ("api_name", "tfinal", "timestamp", "nodeUrl", "success", "reason", "hash") 
+      VALUES (?, ?, ?, ?, ?, ?, ?)
+    `)
+
+    const insertMany = db.transaction((items) => {
```

Comment:
We do insertMany twice which could probably be refactored out to a batch insert function:

`function batchInsert(stmt, items, fields) {
  for (const item of items) {
    stmt.run(...fields.map(field => item[field]));
  }
  stmt.finalize()
}
`